### ISSUE TITLE
Update GUI's progress bar to actually progress when downloading mods

### ISSUE THREAD

**Issue by Nincodedo** [Nincraft/ModPackDownloader#71]

**Comment by Raman1121** [comment]
Hey, I would like to work on this issue.
Can you direct me to the file which needs to be modified? @Nincodedo

**Comment by Nincodedo** [comment]
@Raman1121 In the gui project, the [GUI controller](https://github.com/Nincraft/ModPackDownloader/blob/master/modpackdownloader-gui/src/main/java/com/nincraft/modpackdownloader/gui/Controller.java) has the progressBar and modpackDownloaderManager. The progressBar needs to be updated with how many mods have been downloaded, however there's no way to currently get that information from the manager.

In the core project, the [Reference](https://github.com/Nincraft/ModPackDownloader/blob/master/modpackdownloader-core/src/main/java/com/nincraft/modpackdownloader/util/Reference.java) class has a downloadCount and downloadTotal that keep track of your downloads. This isn't accessible from an instance of the ModpackDownloaderManager so it would need to be refactored.

Let me know if you have any other questions.

**Comment by Raman1121** [comment]
@Nincodedo I understood the problem to some extent. Here is what I was able to think after some initial scanning.

First, we need to access the variables downloadCount and downloadVariable from the Reference class and then using some basic math, we need to display a progress bar (possibly using the JProgressBar class). This we need to do in the Controller.java class.

Also, can you tell me what is the exact function of the variables downloadCount and downloadTotal?

Please tell me if I am thinking in the right direction or not.

**Comment by Nincodedo** [comment]
Yes you are correct. The downloadCount is the number of successfully downloaded mods and downloadTotal is the total number of mods that need to be downloaded. You would basically want to divide downloadCount by downloadTotal to get a percentage to show as the progress bar.

**Comment by Raman1121** [comment]
Awesome!
Can you assign this issue to me, please?

**Comment by Nincodedo** [comment]
Apparently I can't since you don't have access to the repo. I'll assign it to myself even though you are working on it just so everyone knows it is being worked on.

**Comment by Raman1121** [comment]
Is using JProgressBar class to make the progress bar, the correct approach? @Nincodedo

**Comment by Nincodedo** [comment]
Should just be ProgressBar as seen here.

https://github.com/Nincraft/ModPackDownloader/blob/master/modpackdownloader-gui/src/main/java/com/nincraft/modpackdownloader/gui/Controller.java#L16

**Comment by Nincodedo** [comment]
@Raman1121, didn't get around to this?
